src/assert.sh

```diff
@@ -9,6 +9,36 @@ function fail() {
   console_results::print_failure_message "${label}" "$message"
 }
 
+function assert_true() {
+  local actual="$1"
+
+  if [[ "$actual" != "true" && "$actual" != "0" ]]; then
+    local label
+    label="$(helper::normalize_test_function_name "${FUNCNAME[1]}")"
+
+    state::add_assertions_failed
+    console_results::print_failed_test "${label}" "true or 0" "but got " "${actual}"
+    return
+  fi
+
+  state::add_assertions_passed
+}
+
+function assert_false() {
+  local actual="$1"
+
+  if [[ "$actual" != "false" && "$actual" != "1" ]]; then
+    local label
```

Comment:
[nit] Note that this does not seem to work on functions. If I pass `assert_true ls` the assertion fails. A potential problem may be that the concept of true and false as used in `if` statements does not apply with this assertion. If statements really just match the exit code of the expression. Bash makes use of the `test` program to perform much of the boolean logic in BASH, which has the `[[ ... ]]` shorthand.

Future work might want to test if the value passed is a function (`type -f <var>`) or a value and do something. different.

Reply:
I will work on this, many thanks!!